### ISSUE TITLE
Error when saving settings: Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by djdembeck** [Kareadita/Kavita#1966]
**Describe the bug**
When saving settings from the web UI, an error occurs and the settings are not saved.

STR:
1. Go to General settings.
2. Change Logging Level from `Debug` to `Information`.
3. Press 'Save'

Server:
- unRAID version: 6.11.5
- Kavita version: 0.7.2.0

**Additional context**
Installed app from unRAID apps, as the Kavita guide suggests. Standard/default setup, with 1 admin user.

Stack trace:
```
[Kavita] [2023-05-07 17:37:23.920 -05:00  49] [Error] API.Middleware.ExceptionMiddleware There was an exception
System.NullReferenceException: Object reference not set to an instance of an object.
   at API.Controllers.SettingsController.UpdateSettings(ServerSettingDto updateSettingsDto) in /home/runner/work/Kavita/Kavita/API/Controllers/SettingsController.cs:line 195
   at lambda_method1171(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at API.Startup.<>c.<<Configure>b__4_5>d.MoveNext() in /home/runner/work/Kavita/Kavita/API/Startup.cs:line 370
--- End of stack trace from previous location ---
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.ResponseCaching.ResponseCachingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at API.Middleware.ExceptionMiddleware.InvokeAsync(HttpContext context) in /home/runner/work/Kavita/Kavita/API/Middleware/ExceptionMiddleware.cs:line 51
```

**Comment by majora2007** [comment]
Can you clear your cache for me and retry?

**Comment by djdembeck** [comment]
I attempted the STR, on a different browser (first try was Brave/chromium, 2nd Firefox), with the same result.

**Comment by majora2007** [comment]
I'm seeing this myself on docker stable builds only. The non-docker seem to work. I'll get a new image published soon.

**Comment by majora2007** [comment]
A new image has been updated to replace :latest. Please pull it down and validate for me.

**Comment by djdembeck** [comment]
I pulled the new image (`docker pull kizaing/kavita:latest`) and re-created the image. I have the same error as before unfortunately. Tried on both browsers again.

**Comment by djdembeck** [comment]
I found the issue. Kavita isn't verifying or setting file ownership at startup. It looks like it used to, but was disabled at some point: https://github.com/Kareadita/Kavita/blob/ef248fe74393fd6ca1088ba7aea9fd81dc6819de/entrypoint.sh#L35-L50

I had to `chown -R 1001:123 /kavita/config`, then restart the container.

**Comment by majora2007** [comment]
Are you running as root?

**Comment by djdembeck** [comment]
I haven't modified the default docker on unraid in any manner. So whatever the container defaults to

**Comment by therobbiedavis** [comment]
> I haven't modified the default docker on unraid in any manner. So whatever the container defaults to

By default it runs as root, which may not be appropriate for all instances.

Docker allows running containers as different users and groups.

Docker compose:
![image](https://github.com/Kareadita/Kavita/assets/7907265/3c19defa-185e-4ae0-9f1c-6a7ad5423a61)

Docker run:
```
docker run --user 1001:1001 --name kavita -p 5000:5000 \
-d kizaing/kavita:latest \
...
```

**Comment by majora2007** [comment]
Can someone give me the payload from dev tools when they save the settings? 

I'm trying to debug this, but it works on everything but the docker on unraid it seems. 

![image](https://github.com/Kareadita/Kavita/assets/735851/e195f97e-9be4-45ac-ad65-e3ce12ab1170)
![image](https://github.com/Kareadita/Kavita/assets/735851/f354342e-7b90-48c9-8607-47a526b049c2)

**Comment by scare376** [comment]
![opera_Q653WeeEEv](https://github.com/Kareadita/Kavita/assets/2068470/065a6531-b6da-451a-887c-9f04bc1d6301)
![opera_2a519VKADl](https://github.com/Kareadita/Kavita/assets/2068470/73d5d499-0a8d-4265-9790-87a569f8838e)

**Comment by majora2007** [comment]
We were able to finally solution this. It is currently fixed on v0.7.2.11. If you are on stable (:latest), then please wait as we work to provide a solution for that branch.
